### PR TITLE
Update SCP Workflow REST API with Destination retrieval change

### PR DESCRIPTION
**Pull request by KavithaSiva**
## What Has Changed?

There is a new API `ScpCfServiceDestinationLoader.getDestinationForService` introduced
that creates a `Destination` programmatically using values read from `VCAP_SERVICES` for Workflow service.
PR: https://github.wdf.sap.corp/MA/sdk/pull/4847

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- ~[x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.~

### CODE REVIEW

File: docs/java/features/rest/clients/scp-workflow-rest-api.mdx [hunk 1]
```diff
@@ -135,75 +135,10 @@ applications:
       - route: <omitted-on-purpose>
 ```
 
+This ensures that the `VCAP_SERVICES` environment variable of the CF application gets enhanced with an additional entry containing the connection details of the
+newly bound workflow service.
```
Comment (artemkovalyov) [inline docs/java/features/rest/clients/scp-workflow-rest-api.mdx:139]:
```suggestion
This ensures that the `VCAP_SERVICES` environment variable of the CF application gets enhanced with an additional entry containing the connection details of the newly bound workflow service.
```

File: docs/java/features/rest/clients/scp-workflow-rest-api.mdx [hunk 2]
```diff
@@ -248,14 +183,26 @@ After adding the dependency to your `pom.xml` file run `mvn clean install` to le
...
+final HttpDestination destination =
+                ScpCfServiceDestinationLoader.getDestinationForService(
+                        ScpCfServiceDestinationLoader.CfServices.WORKFLOW,
+                        "my-workflow");
 ```
 
+:::caution
+The `ScpCfServiceDestinationLoader.getDestinationForService` API currently only works out of the box for the workflow service.
+For other services, the alternative would be to create a Destination manually in CF using values in the `VCAP_SERVICES` and then accessing the
+destination using the `DestinationAccessor`. For details refer to [Additional information](scp-workflow-rest-api#additional-information) section.
```
Comment (artemkovalyov) [inline docs/java/features/rest/clients/scp-workflow-rest-api.mdx:201]:
```suggestion
For other services, the alternative would be to create a Destination manually in CF using values in the `VCAP_SERVICES` and then accessing the destination using the `DestinationAccessor`.
For details refer to [Additional information](scp-workflow-rest-api#additional-information) section.
```

File: docs/java/features/rest/clients/scp-workflow-rest-api.mdx [hunk 3]
```diff
@@ -306,6 +253,98 @@ The Java client library for SCP Workflow enables the developer to:
   The SAP Cloud Platform Neo landscape in **Not supported!**
 - In the current state, it is required to create a destination manually instead of letting the library resolve it for you via VCAP_SERVICES binding available on SCP Cloud Foundry.
 
+## Additional Information
+
+:::info
+Please note the steps outlined in the below section are only required if you chose to skip using the convenience API `ScpCfServiceDestinationLoader.getDestinationForService`
+introduced [here](scp-workflow-rest-api#create-destination) and instead would like to use `DestinationAccessor` for e.g while trying to use generated clients of other services.
```
Comment (artemkovalyov) [inline docs/java/features/rest/clients/scp-workflow-rest-api.mdx:260]:
```suggestion
Please note the steps outlined in the below section are only required if you chose to skip using the convenience API `ScpCfServiceDestinationLoader.getDestinationForService` introduced [here](scp-workflow-rest-api#create-destination) and instead would like to use `DestinationAccessor` for e.g. while trying to use generated clients of other services.
```

File: docs/java/features/rest/clients/scp-workflow-rest-api.mdx [hunk 4]
```diff
@@ -306,6 +253,98 @@ The Java client library for SCP Workflow enables the developer to:
...
+:::
+
+### Creating HTTP Destinations Manually
+
+To create HTTP destinations manually in CF from the values read from `VCAP_SERVICES` of the CF application, please follow the below steps.
+The steps below can be continued in this example from [Bind your App to Service Instance](scp-workflow-rest-api#bind-your-app-to-service-instance) section.
```
Comment (artemkovalyov) [inline docs/java/features/rest/clients/scp-workflow-rest-api.mdx:266]:
```suggestion
To create HTTP destinations manually in CF from the values read from `VCAP_SERVICES` of the CF application, please follow the below steps.
The steps below can be continued in this example from the [Bind your App to Service Instance](scp-workflow-rest-api#bind-your-app-to-service-instance) section.
```

File: docs/java/features/rest/clients/scp-workflow-rest-api.mdx [hunk 5]
```diff
@@ -306,6 +253,98 @@ The Java client library for SCP Workflow enables the developer to:
...
+
+Once the app is bound to the workflow service (here in our example) and app deployment has finished, go to your CF space and navigate to `Services\Service Instances`.
+You should see the service instance you created along with the information that is bound to your application.
+
+Click on the service instance name, for instance `my-workflow`, in the upcoming screen you should see the headline `Service Instance: my-workflow - Referencing Apps`.
+Make sure that the entry belongs to your app is selected in the table below, given that multiple apps are bound to the same service instance.
+
+Consider the `JSON` content below the table.
+For your convenience, we recommend copying that `JSON` to a text editor.
+Here is a quick example:
```
Comment (artemkovalyov) [inline docs/java/features/rest/clients/scp-workflow-rest-api.mdx:278]:
```suggestion
Once the app is bound to the workflow service (here in our example) and app deployment has finished, go to your CF space and navigate to `Services\Service Instances`.
You should see the service instance you created along with the information that is bound to your application.

Click on the service instance name, for instance `my-workflow`, in the upcoming screen you should see the headline `Service Instance: my-workflow - Referencing Apps`.
Make sure that the entry that belongs to your app is selected in the table below, given that multiple apps are bound to the same service instance.

Consider the `JSON` content below the table.
For your convenience, we recommend copying that `JSON` to a text editor.
Here is a quick example:
```

**Review by artemkovalyov — CHANGES_REQUESTED** [review]
Looks good. Just minor corrections.

**Comment by artemkovalyov** [comment]
@KavithaSiva,  I think you've had troubles with Vale. That's where a couple of weird line-breaks come from. Usually, the only thing to make fix it is to make sure each sentence is on a separate line. It should be also fixed by the prettier liner. You can invoke it with `npm run prettier`.

**Review by artemkovalyov — APPROVED** [review]
I fixed and committed the minors. This can be merged.
If you need a second eye from another colleague, please, arrange it.
Otherwise can be merged.

**Comment by KavithaSiva** [comment]
> @KavithaSiva, I think you've had troubles with Vale. That's where a couple of weird line-breaks come from. Usually, the only thing to make fix it is to make sure each sentence is on a separate line. It should be also fixed by the prettier liner. You can invoke it with `npm run prettier`.

Yes, some of my lines were on the same line that caused `Vale` to complain. I had run `npm run prettier` but that hadn't fixed these issues reported by Vale. I ended up correcting them manually, wasn't too much of a problem.

**Comment by KavithaSiva** [comment]
> I fixed and committed the minors. This can be merged.
> If you need a second eye from another colleague, please, arrange it.
> Otherwise can be merged.

No, as it is just a minor update to the API, I think your review should suffice, hence I am merging this.

**Pull request merged**